### ISSUE TITLE
gemato fails to find top level Manifest

### ISSUE THREAD

**Issue by bearsh** [mgorny/gemato#5]
after `emerge --sync` or by calling gemato directly (`gemato verify /usr/portage/`) I get:
`ERROR:root:Top-level Manifest not found in /usr/portage/`
I skimmed though `find_top_level.py` but didn't get the logic how it is supposed to work. My suspicion is, it is because /usr/portage/ is a overlayfs:
```bash
$ mount
overlay on /usr/portage type overlay (rw,relatime,lowerdir=./portree_ro,upperdir=./portree_tmpfs/upper,workdir=./portree_tmpfs/work)
```
and therefor
`stat  /usr/portage` and `stat /usr/portage/Manifest` return a different device

**Comment by mgorny** [comment]
I'm aware of the problem. This is basically a major design issue of overlayfs and I don't see any obvious fix. Any tool that's aware of filesystem boundaries (`find -x`, `git`...) won't work on it.

**Comment by bearsh** [comment]
what about a way to specify the to level manifest from the outside (cmd arg)?

**Comment by mgorny** [comment]
Gemato enforces filesystem boundaries both ways. The logic in finding Manifest is just auxiliary to match the main logic that refuses escaping the filesystem while verifying.

**Comment by bearsh** [comment]
while you may have your reasons for your desing choices, from a user point of view nothing escapes the filesystem as the whole repo is in that specific filesystem.

**Comment by mgorny** [comment]
Except it isn't because you're having a mix of two separate filesystems.

**Comment by mgorny** [comment]
This can now be disabled explicitly since 872fdd1495623b90054c420c84b2b3cb0be7c61f (except there's no CLI option for it). I will look into disabling it by default but this has security implications, so needs a wider agreement first.

**Comment by bearsh** [comment]
:+1: 
thx!

**Comment by bearsh** [comment]
seems to be fixed in 952fdcf63a227c341c96206d505be5c44546dd5b, closing it...
